addons/mail/static/src/composer/suggestion_service.js

```diff
@@ -142,7 +142,42 @@ export class SuggestionService {
                 }
             }
         }
-        const sortFunc = (p1, p2) => {
+        return {
+            type: "Partner",
+            mainSuggestions: sort
+                ? this.sortPartnerSuggestions(mainSuggestionList, thread, cleanedSearchTerm)
+                : mainSuggestionList,
+            extraSuggestions: sort
+                ? this.sortPartnerSuggestions(extraSuggestionList, thread, cleanedSearchTerm)
+                : extraSuggestionList,
+        };
+    }
+
+    sortPartnerSuggestions(partners, thread = undefined, searchTerm = "") {
```

Comment:
`searchTerm` looks more important that `thread`.
In all uses of this function you pass `searchTerm` but not necessarily `thread`. So `thread` should be after `searchTerm`.